lib/iris/_lazy_data.py

```diff
@@ -147,9 +147,14 @@ def lazy_masked_fill_value(data):
     # If lazy, get the smallest slice of the data from which we can retrieve
     # the fill_value.
     if is_lazy_data(data):
-        inds = tuple([0] * (data.ndim-1))
-        smallest_slice = data[inds][:0]
-        data = as_concrete_data(smallest_slice)
+        inds = [0] * data.ndim
```

Comment:
Functionally quite different though - for NetCDF that could mean loading the entire dataset and then discarding all the actual values.